File: src/main/java/dev/abstratium/partner/entity/PartnerDiscriminator.java
package dev.abstratium.partner.entity;

/**
 * Constants for partner type discriminator values.
 * Used in JPA @DiscriminatorValue annotations and throughout the application.
 */
public final class PartnerDiscriminator {
    
    public static final String NATURAL_PERSON = "NATURAL_PERSON";
    public static final String LEGAL_ENTITY = "LEGAL_ENTITY";
    
    private PartnerDiscriminator() {
        // Utility class - prevent instantiation
    }
}

File: src/main/webui/src/app/models/partner-discriminator.ts
/**
 * Constants for partner type discriminator values.
 * Must match the values in Java PartnerDiscriminator class.
 */
export const PartnerDiscriminator = {
  NATURAL_PERSON: 'NATURAL_PERSON',
  LEGAL_ENTITY: 'LEGAL_ENTITY'
} as const;

export type PartnerDiscriminatorType = typeof PartnerDiscriminator[keyof typeof PartnerDiscriminator];
